src/routes/safe/components/Apps/components/ConfirmTxModal/ReviewConfirm.tsx

```diff
@@ -112,6 +118,12 @@ export const ReviewConfirm = ({
   }
 
   const confirmTransactions = (txParameters: TxParameters, delayExecution: boolean) => {
+    const trackData = loadFromStorage<AppTrackData>(APPS_DASHBOARD) || {}
+    let currentTxCount = trackData[currentApp.id]?.txCount
+    saveToStorage(APPS_DASHBOARD, {
+      ...trackData,
+      [currentApp.id]: { ...trackData[currentApp.id], txCount: currentTxCount ? ++currentTxCount : 1 },
+    })
```

Comment:
Please put all the tracking related code into a separate module and import it here and in the app frame.